vowpalwabbit/vw.h

```diff
@@ -47,21 +47,21 @@ vw* initialize(std::unique_ptr<config::options_i, options_deleter_type> options,
     bool skip_model_load = false, trace_message_t trace_listener = nullptr, void* trace_context = nullptr);
 vw* initialize(config::options_i& options, io_buf* model = nullptr, bool skip_model_load = false,
     trace_message_t trace_listener = nullptr, void* trace_context = nullptr);
-vw* initialize(std::string s, io_buf* model = nullptr, bool skip_model_load = false,
+vw* initialize(const std::string& s, io_buf* model = nullptr, bool skip_model_load = false,
```

Comment:
These changes break the interface (definitely the ABI, possibly the API in some cases). Does that matter?

Reply:
Any `std::string` passed to this function will work as before (just without copying!) and we certainly do not guarantee any sort of ABI here. So I think its fine